Added ReplaceIRVar.cpp

ayush-qubit opened the pull request:
Added a file ReplaceIRVar.cpp that can change IR temp variable to Source Variable if "-g" option is enable else replace to "i".

reSHARMA commented on include/Utils/ReplaceIRVar.h:
```diff
@@ -0,0 +1,23 @@
+class ReplaceIRVar{
+private:
+	std::string NewName;
+	std::map<std::string,std::string> HashMap;
```
Can you please add a comment about what are the keys and values in this HashMap?

reSHARMA commented on lib/Utils/ReplaceIRVar.cpp:
```diff
@@ -0,0 +1,82 @@
+#include <iostream>
```
Do you need iostream?

reSHARMA reviewed (changes requested):
Many thanks! Can you remove changes to the .gitignore and readme.md in this PR as they do not fit here :) 
Everything else looks great! Added a few minor comments.

ayush-qubit commented:
But I put "build" directory in .gitignore so that it cannot push my own
"build" directory. And in Readme I have added sudo command when installing
Spatial library, I can remove sudo in readme.

On Sat, 20 Feb 2021 at 21:28, Reshabh Sharma <notifications@github.com>
wrote:

> *@reSHARMA* requested changes on this pull request.
>
> Many thanks! Can you remove changes to the .gitignore and readme.md in
> this PR as they do not fit here :)
> Everything else looks great! Added a few minor comments.
> ------------------------------
>
> In include/Utils/ReplaceIRVar.h
> <https://github.com/reSHARMA/spatial/pull/6#discussion_r579673609>:
>
> > @@ -0,0 +1,23 @@
> +#ifndef REPLACEIRVAR_H
> +#define REPLACEIRVAR_H
> +
> +#include "llvm/IR/Function.h"
> +#include "llvm/IR/BasicBlock.h"
> +#include "llvm/IR/Module.h"
> +#include <iostream>
> +#include <map>
> +
> +class ReplaceIRVar{
> +private:
> +	std::string NewName;
> +	std::map<std::string,std::string> HashMap;
>
> Can you please add a comment about what are the keys and values in this
> HashMap?
> ------------------------------
>
> In lib/Utils/ReplaceIRVar.cpp
> <https://github.com/reSHARMA/spatial/pull/6#discussion_r579673665>:
>
> > @@ -0,0 +1,82 @@
> +#include <iostream>
>
> Do you need iostream?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/reSHARMA/spatial/pull/6#pullrequestreview-594763220>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AI5WIGY7P5S7YJ6T5NJFLNTS77L3RANCNFSM4X57NAPQ>
> .
>


-- 
*Ayush Sharma*
